libexec/add_key.rb

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env ruby
+
+require 'etc'
+require 'fileutils'
+
+case ARGV.length
+when 0
+  raise "Missing the USER and KEY arguments"
+when 1
+  raise "Missing the KEY argument"
+end
+
+# Extract the arguments and get the user's details
+USER = ARGV[0]
+KEY = ARGV[1]
+PASSWD = Etc.getpwnam(USER)
+
+# Become the user
+Process.groups = []
+Process.gid = PASSWD.gid
+Process.egid = PASSWD.gid
+Process.initgroups(USER, PASSWD.gid)
+Process.uid = PASSWD.uid
+Process.euid = PASSWD.uid
+Process.setsid
+ENV['HOME'] = PASSWD.dir
+ENV['USER'] = USER
+ENV['LOGNAME'] = USER
+
+# Ensure authorized_keys exists
+PATH = File.expand_path('~/.ssh/authorized_keys')
+FileUtils.mkdir_p File.dirname(PATH)
+FileUtils.touch PATH
```

Comment:
What permissions does this file get created with?

Reply:
It is now made with `0600`:

https://github.com/openflighthpc/flight-console-api/pull/16/commits/51064b42a065ff48ab0f82a8d6d4ec81f96e1ce9